### ISSUE TITLE
Datainterpreter Message history, available for usage like chat?

### ISSUE THREAD

**Issue by krish240574** [geekan/MetaGPT#1167]
**Feature description**

Is the message history for DI available , like a chat feature? I need to use it in a web app, so the user is updated frequently. 
The history is dumped to data/<timestamp>/plan.json, I see that, but that happens only after the run completes.
Are running messages available? 

I see inside metagpt/roles/role.py, the following :
def get_memories(self, k=0) -> list[Message]:
        """A wrapper to return the most recent k memories of this role, return all when k=0"""
        return self.rc.memory.get(k=k)
and a variable called working_memory

Can they be useful somehow?
        

**Your Feature**

**Comment by geekan** [comment]
There are some codes that have not yet been merged into the open source version, perhaps u'll need to wait for serveral weeks.

**Comment by krish240574** [comment]
Any updates on this, please ? I'd love to be able to echo messages to the user, as the DI executes. For now, the user has to wait till all execution completes, then I can retrieve the dumped history from save_history and send to user. It would be great if I could access regular updates.
